### ISSUE TITLE
Use UserDefinedType for UUIDType and set the server_default

### ISSUE THREAD

**Issue by mehcode** [kvesteri/sqlalchemy-utils#119]
Did a quick test and this works.

``` python
class UUID(sa.types.UserDefinedType):
    def get_col_spec(self):
        return 'UUID DEFAULT uuid_in(md5(random()::text || now()::text)::cstring)'
```

I'm tired of seeing `default=uuid.uuid4` all over my `UUIDType` usage. :laughing: 

@kvesteri What do you think?

**Comment by kvesteri** [comment]
I've had the same problem as you (having to type default=uuid.uuid4 all over the place). However I think this is not the way to go for the following reasons:
* This is PostgreSQL specific
* What if user wants to use uuid-ossp for PostgreSQL (see http://stackoverflow.com/a/13146662)? 
* What if user wants to override the default?

I think the best way to implement would be to introduce some kind of type rule system. The rules would be implemented using listeners.
```
@type_rule(UUIDType)
def process_uuid(column):
    if column.default is None:
        column.default = uuid.uuid4
```

**Comment by jace** [comment]
After googling around a bit, I've gathered UUID1 with random multicast MAC ids are the best choice as timestamped UUIDs are better for index performance, and randomised MAC ids address the security concerns. The `uuid_generate_v1mc()` function from PostgreSQL's `uuid-ossp` does this, while Python's `uuid.UUID1()` does not randomise MAC ids.

I'm trying this:

```python
from sqlalchemy.sql.functions import GenericFunction

class uuid_generate_v1mc(GenericFunction):
    type = sqlalchemy.dialects.postgresql.UUID
```

This lets me use `func.uuid_generate_v1mc()`:

```python
class MyModel(Base):
    id = Column(UUIDType(binary=False), default=func.uuid_generate_v1mc(), primary_key=True)
```

However, since this is now limited to PostgreSQL, I tried wrapping it into a function that can be used for `default`:

```python
def default_uuid1(context):
    if context.dialect.name == 'postgresql':
        return func.uuid_generate_v1mc()
    else:
        return uuid.uuid1()
```

This no longer works because the default function is now expected to return a value, not a SQL function. I'm guessing I can do some dialect-based customisation on the function and return an actual value for non-PostgreSQL dialects. I don't know how to work around this, but I found a pure-Python implementation that works:

```python
import uuid

def uuid1mc():
    return uuid.uuid1(node=uuid._random_getnode())
```

I'll be nice to be able to push this into the database as a `server_default` however.
